data/helpers.d/psql

```diff
@@ -1,148 +1,273 @@
-# | arg: user - Owner of the database
-# | arg: name - Name of the database
-# | arg: pwd - Password of the database. If not given, a password will be generated
-ynh_psql_setup_db () {
-	local db_user="$1"
-	local db_name="$2"
-	local new_db_pwd=$(ynh_string_random)	# Generate a random password
-	# If $3 is not given, use new_db_pwd instead for db_pwd.
-	local db_pwd="${3:-$new_db_pwd}"
-	ynh_psql_create_db "$db_name" "$db_user" "$db_pwd"	# Create the database
-	ynh_app_setting_set "$app" psqlpwd "$db_pwd"	# Store the password in the app's config
+	ynh_psql_connect_as --user="postgres" --password="$(sudo cat $PSQL_ROOT_PWD_FILE)" \
+		--database="$database" <"$file"
 }
 
-# Create a database and grant privilegies to a user
+# Create a database and grant optionnaly privilegies to a user
+#
+# [internal]
 #
 # usage: ynh_psql_create_db db [user [pwd]]
 # | arg: db - the database name to create
 # | arg: user - the user to grant privilegies
-# | arg: pwd  - the user password
+# | arg: pwd - the password to identify user by
 ynh_psql_create_db() {
-	local db="$1"
-	local user="$2"
-	local pwd="$3"
-	ynh_psql_create_user "$user" "$pwd"
-	sudo --login --user=postgres createdb --owner="$user" "$db"
+	local db=$1
+
+    local sql="CREATE DATABASE ${db};"
+
+    # grant all privilegies to user
+	if [[ $# -gt 1 ]]; then
+		sql+="GRANT ALL PRIVILEGES ON DATABASE ${db} TO ${2} WITH GRANT OPTION;"
+	fi
+
+	ynh_psql_execute_as_root --sql="$sql"
 }
 
 # Drop a database
 #
+# [internal]
+#
+# If you intend to drop the database *and* the associated user,
+# consider using ynh_psql_remove_db instead.
+#
 # usage: ynh_psql_drop_db db
 # | arg: db - the database name to drop
-# | arg: user - the user to drop
-ynh_psql_remove_db() {
-	local db="$1"
-	local user="$2"
-	sudo --login --user=postgres dropdb "$db"
-	ynh_psql_drop_user "$user"
+ynh_psql_drop_db() {
+	local db=$1
+	sudo --login --user=postgres dropdb $db
 }
 
 # Dump a database
 #
 # example: ynh_psql_dump_db 'roundcube' > ./dump.sql
 #
-# usage: ynh_psql_dump_db db
-# | arg: db - the database name to dump
+# usage: ynh_psql_dump_db --database=database
+# | arg: -d, --database - the database name to dump
 # | ret: the psqldump output
 ynh_psql_dump_db() {
-	local db="$1"
-	sudo --login --user=postgres pg_dump "$db"
-}
+	# Declare an array to define the options of this helper.
+	local legacy_args=d
+	declare -Ar args_array=([d]=database=)
+	local database
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
 
+	sudo --login --user=postgres pg_dump "$database"
+}
 
 # Create a user
 #
+# [internal]
+#
 # usage: ynh_psql_create_user user pwd [host]
 # | arg: user - the user name to create
+# | arg: pwd - the password to identify user by
 ynh_psql_create_user() {
-	local user="$1"
-	local pwd="$2"
-	sudo --login --user=postgres psql -c"CREATE USER $user WITH PASSWORD '$pwd'" postgres
+	local user=$1
+	local pwd=$2
+	ynh_psql_execute_as_root --sql="CREATE USER $user WITH ENCRYPTED PASSWORD '$pwd'"
```

Comment:
Did you try it ?
Sure $pwd will be interpreted before going to ynh_psql_execute_as_root ?

Reply:
Yep, tested. Here a log of my ynh-dev:

```
2019-02-20 23:32:06,069: DEBUG - + ynh_psql_create_user lstu w3i9QOs6zW1jPW1OUlsiyg0z
2019-02-20 23:32:06,069: DEBUG - + local user=lstu
2019-02-20 23:32:06,069: DEBUG - + local pwd=w3i9QOs6zW1jPW1OUlsiyg0z
2019-02-20 23:32:06,069: DEBUG - + ynh_psql_execute_as_root '--sql=CREATE USER lstu WITH ENCRYPTED PASSWORD '\''w3i9QOs6zW1jPW1OUlsiyg0z'\'''
```

Reply:
I trust you if you have tested it.